swan-lake/learn-the-platform/test-document-the-code/test-ballerina-code/test-services-and-clients.md

```diff
@@ -110,6 +110,31 @@ function getMockResponse() returns http:Response {
 }
 ```
 
+### Mock `final` clients
+
+Object mocking cannot be used as final clients cannot be modified. To facilitate testing, it is recommended 
+to write the client initialization logic in a separate function and assign the returned value to the client. 
+This initialization function can then be mocked using the legacy function mocking feature.
```

Comment:
What about using `compile-time` function mocking feature instead of `legacy`?

Reply:
+1